noria-server/mir/src/node.rs

```diff
@@ -485,8 +511,9 @@ impl MirNodeType {
             } => {
                 group_by.push(c);
             }
-            MirNodeType::Filter { ref mut conditions } => {
-                conditions.push(None);
+            MirNodeType::Filter { .. } => {}
```

Comment:
Is it intentional that this no longer does a `conditions.push`?

Reply:
Yep! The new representation of conditions only adds an element of (index, condition) if the condition is present, instead of having either Some(condition) or None for every index. See Malte's note above about why I changed the representation.

I read your comments -- thank you so much for the incredibly detailed feedback! I haven't used Rust enough to be familiar with the conventions, so it's very helpful -- but I probably won't get a chance to make the changes until after Weds.